### ISSUE TITLE
Cursor autohide feature

### ISSUE THREAD

**Issue by ysoloviov** [lxqt/qterminal#403]
Can this feature be added? It was very useful when I used vte based terminals. Konsole also has this feature but for my purposes it is a little bit bloated.

**Comment by RomanBruhanov** [comment]
There is setAutoHideCursor in qtermwidget, but for that to work you will need to link to KDE. I guess it will be a lot of work to implement hiding in lxqt natively. Easier to stick with KDE or GTK terminals.

**Comment by yan12125** [comment]
KWidgetsAddons is also in Tier 1, so depending on it will not be a big problem. Of course it's better to be optional to avoid issue on platforms where KF5 is not fully supported (e.g., macOS).

**Comment by tsujan** [comment]
> KWidgetsAddons is also in Tier 1

Yes tier-1 K-things are nice extensions of Qt that don't depend on KDE. I think we could use them in LXQt if really needed -- as we use `KWindowSystem`.

**Comment by luebking** [comment]
Why does that need some KDE stuff?
You monitor MouseMove events re/start a timer and when that actually fires QApplication::setOverrideCursor(Qt::BlankCursor); and QApplication::restoreOverrideCursor(); once the mouse starts moving.

Alternatively (just seeing this in phototonic) you could poll the mouse position (not sure whether I'm gonna replace that during the alpha cycle, but I'm pretty sure that's gonna leave there at some point ;)

**Comment by tsujan** [comment]
> Why does that need some KDE stuff?

You're right: there's no need to any KDE stuff for that.

At the time this report was opened, I was rather a bystander (and a user) here. Months ago, I started to maintain qterminal and qtermwidget, went through their bug trackers to find the most important reports, and fixed them. Sadly, I can't check all valid reports, being busy with several other codes. Contributions are really appreciated.

**Comment by luebking** [comment]
Preferably in qterminal or qtermwidget?
(I'm gonna check whether the latter tracks the mouse anyway and would base the decision whether to go for polling or event handling based on that - the implementation can be in either location)

**Comment by tsujan** [comment]
> Preferably in qterminal or qtermwidget?

Wherever you see fit — I'm confident of your coding skills, having seen your codes and used some of them in [Kvantum](https://github.com/tsujan/Kvantum).

`qtermwidget` is the place where we put the *core* codes, i.e., those that can be used by any app based on `qtermwidget`. QTerminal is only one app that uses `qtermwidget`. So, if the feature you want to implement is for QTerminal, `qterminal` is the proper place.

**Comment by marcusbritanicus** [comment]
> So, if the feature you want to implement is for QTerminal, qterminal is the proper place.

@tsujan It will be hassle to implement it in qterminal. The mouse should be hidden only when it's inside the qtermwidget. So I think it is better to implement the `setCursorAutohide` in qtermwidget and turn that on in qterminal with the help of a switch.

**Comment by tsujan** [comment]
> @tsujan It will be hassle to implement it in qterminal.

It's OK with me. What can I say when two competent devs agree on something?
